core/store/models/common.go

```diff
@@ -183,32 +183,51 @@ func (j JSON) Bytes() []byte {
 	return []byte(j.String())
 }
 
-// Add returns a new instance of JSON with the new value added.
-func (j JSON) Add(insertKey string, insertValue interface{}) (JSON, error) {
+// AsMap returns j as a map
+func (j JSON) AsMap() (map[string]interface{}, error) {
 	output := make(map[string]interface{})
-
 	switch v := j.Result.Value().(type) {
 	case map[string]interface{}:
 		for key, value := range v {
-			if key != insertKey {
-				output[key] = value
-			}
+			output[key] = value
 		}
-		output[insertKey] = insertValue
 	case nil:
-		output[insertKey] = insertValue
 	default:
-		return JSON{}, errors.New("can only add to JSON objects or null")
+		return nil, errors.New("can only add to JSON objects or null")
 	}
+	return output, nil
+}
 
-	bytes, err := json.Marshal(output)
+// NewJSONFromMap returns m as a JSON object, or errors
+func NewJSONFromMap(m map[string]interface{}) (JSON, error) {
+	bytes, err := json.Marshal(m)
 	if err != nil {
 		return JSON{}, err
 	}
-
 	return JSON{Result: gjson.ParseBytes(bytes)}, nil
 }
 
+// jsonObject is used to pass multiple key/value pairs to JSON
+// object-manipulation utilities
+type jsonObject map[string]interface{}
+
+// Add returns a new instance of JSON with the new value added.
+func (j JSON) Add(insertKey string, insertValue interface{}) (JSON, error) {
+	return j.MultiAdd(jsonObject{insertKey: insertValue})
+}
+
+// MultiAdd returns a new instance of j with the new values added.
+func (j JSON) MultiAdd(keyValues jsonObject) (JSON, error) {
+	output, err := j.AsMap()
+	if err != nil {
+		return JSON{}, err
+	}
+	for key, value := range keyValues {
+		output[key] = value
+	}
+	return NewJSONFromMap(output)
+}
```

Comment:
What was trying to be done by having a public method task a private value?

Reply:
Is this function heavily used in later PRs?  I don't see much use case of it in this PR, and it could be done with multiple Add()'s.

Reply:
Also, another option would be to add a JSON.Merge(JSON) function which would be more generalized.